### DISCUSSION TITLE
Efficient KAN for supervised learning.

### DISCUSSION THREAD

**Discussion by AhalyaSubramanian** [Blealtan/efficient-kan#68] [Q&A] [answered]
Hi, Efficient KAN is Awesome. I need to implement Efficient KAN for the below dataset and need to perform Supervised Learning. My dataset comprises of 2700 rows and 100 columns. First column (class) is the target or label and rest columns consist of (x,y,z) co-ordinate for 33 joints in human body. I have attached the dataset for your kind reference. How could I used **EfficientKAN** for performing supervised learning over the given dataset. Kindly guide me so that I will be going in right path. Thank You !!!

[dataset_all_points.csv](https://github.com/user-attachments/files/20758706/dataset_all_points.csv)

**Comment by DanteTrb** [comment, marked as answer]
👋 Hi @AhalyaSubramanian,

You're working on a very interesting dataset—33 joints with full (x, y, z) spatial representation per sample is a rich format for supervised learning!

I've inspected your file dataset_all_points.csv:

2700 samples (rows)

99 features (33 joints × 3 coordinates)

1 target column named class

To use EfficientKAN for classification, here's a minimal and clean pipeline:

```
import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn.metrics import classification_report
from efficientkan.model import KAN

# Load and split
df = pd.read_csv("dataset_all_points.csv")
X = df.drop("class", axis=1).values
y = df["class"].values
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

# Train EfficientKAN
model = KAN(input_dim=X.shape[1], output_dim=len(set(y)), task="classification")
model.fit(X_train, y_train, epochs=100, batch_size=32)

# Evaluate
y_pred = model.predict(X_test)
print(classification_report(y_test, y_pred))
```

✅ This setup assumes efficientkan is properly installed (see the repo instructions).
✅ You might consider normalizing your input data depending on the scale of your coordinates.

Let me know if you'd like a notebook or further help, happy to refine this together!

Cheers and best of luck with your project!
(PS: If this helped you get unstuck, feel free to mark it as accepted so others can find it easily 😊)

**Reply by AhalyaSubramanian** [reply]
Thank you for your great support !!!

**Reply by DanteTrb** [reply]
Really glad it helped!🫡
If everything works fine now, feel free to mark the thread as answered, so future users with similar questions can find it more easily.
Cheers and best of luck with your work with KAN!🤗
